### PR TITLE
Remove the unnecessary parameter `dataType` in `resolveColumnVector` method

### PR DESCRIPTION
**Pull request by firestarman**
This fixes https://github.com/NVIDIA/spark-rapids/issues/4339

Signed-off-by: Firestarman <firestarmanllc@gmail.com>

### CODE REVIEW

**Comment by firestarman** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuExpandExec.scala [hunk 1]
```diff
@@ -163,7 +163,7 @@ class GpuExpandIterator(
         val (cv, nullColumnReused) = expr.columnarEval(cb) match {
           case null => getOrCreateNullCV(sparkType)
           case other =>
-            (GpuExpressionsUtils.resolveColumnVector(other, cb.numRows, sparkType), false)
+            (GpuExpressionsUtils.resolveColumnVector(other, cb.numRows), false)
```
Comment (GaryShen2008) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuExpandExec.scala:166]: Does the return value "other" of expr.columnarEval() have the same type as the expr.dataType?

**Comment by sperlingxx** [comment]
LGTM

**Review by sperlingxx — APPROVED** [review]
LGTM

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuExpandExec.scala (hunk 1 above)
Comment (sperlingxx) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuExpandExec.scala:166]: As long as the expression itself is correctly programmed, sparkType is the result type of the "expr".

Comment (firestarman) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuExpandExec.scala:166]: Yes. You can find it here
https://github.com/NVIDIA/spark-rapids/blob/branch-22.02/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuExpandExec.scala#L162

**Pull request merged**